numpy/_core/src/multiarray/multiarraymodule.c

```diff
@@ -1609,7 +1652,11 @@ _array_fromobject_generic(
 
         /* One more chance for faster exit if user specified the dtype. */
         oldtype = PyArray_DESCR(oparr);
-        if (PyArray_EquivTypes(oldtype, dtype)) {
+        unsigned char viewable = PyArray_ViewableTypes(oldtype, dtype);
```

Comment:
The equiv cast may not matter, but then a single field struct to non-struct could return views... not today, I guess